pacsman/utils.py

```diff
@@ -73,3 +77,13 @@ def dataset_attribute_fetcher(dataset, data_attribute):
     except AttributeError:
         # Dataset has a bug where it ignores the default=None when getattr is called.
         return None
+
+def dicom_file_iterator(folder: str) -> Iterable[Dataset]:
+    for root, dirs, files in os.walk(folder):
+        for file in files:
+            dicom_file = os.path.join(root, file)
+            dataset = dcmread(dicom_file)
+            yield dataset
+
+def dicom_filename(dataset: Dataset) -> str:
+    return f'{dataset.SOPInstanceUID}.dcm'
```

Comment:
Using `walk` instead of `scandir` enables loading nested hierarchies (`test_dicom_data` isn't one but I've set another one up already for emass), but that also increases the risk of getting other non-DICOM files mixed in. Maybe there should be an extension check or a `try` around `dcmread` here?